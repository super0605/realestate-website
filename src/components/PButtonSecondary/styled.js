import styled from "styled-components";
import { Button, Typography } from "antd";

const { Text } = Typography;

export const CustomButton = styled(Button)`
  /* generel style */
  height: ${({ height }) => height};
  width: ${({ width }) => width};
  border: 0px;
  display: flex;
  padding: 0;
  align-items: center;
  background-color: transparent;
  box-shadow: unset;

  /* hover, focus, active style */
  &.ant-btn[disabled] {
    background-color: transparent;
    background: transparent;
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.color};
      opacity: 0.4;
    }
  }
  &.ant-btn[disabled]:hover {
    background-color: transparent;
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.color};
    }
  }
  &.ant-btn[disabled]:focus {
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.color};
    }
  }

  &:hover {
    background-color: transparent;
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.hoverColor};
    }
  }
`;

export const ButtonText = styled(Text)`
  ${({ theme }) => theme.components.button.buttonSecondary.textStyle}
  color: ${({ theme, ptype, pchecked }) =>
    pchecked === "true"
      ? theme.colors.primary
      : theme.components.button.buttonSecondary[ptype].color};
`;

export const SecondaryWrapper = styled(Text)`
  padding: 5px;
  ${({ theme }) => theme.components.button.buttonSecondary.textStyle}
  color: ${({ theme, ptype }) => theme.components.button.buttonSecondary[ptype].color};
`;
