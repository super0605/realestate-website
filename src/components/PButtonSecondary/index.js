import React, { useState } from "react";
import PropTypes from "prop-types";
import PCheckbox from "../PCheckbox";
import { CustomButton, ButtonText, SecondaryWrapper } from "./styled";

const PButtonSecondary = ({
  pname,
  ptype,
  psecondary,
  psecondaryAlign,
  pwithCheck,
  pchecked,
  width,
  height,
  children,
  ...rest
}) => {
  const [checked, setChecked] = useState(pchecked);
  const onClick = () => {
    pwithCheck && setChecked(!checked);
  };

  return (
    <CustomButton
      ptype={ptype}
      psecondary={psecondary}
      psecondaryalign={psecondaryAlign}
      pwithcheck={pwithCheck.toString()}
      pchecked={checked.toString()}
      height={height}
      width={width}
      onClick={onClick}
      {...rest}
    >
      {children ? (
        children
      ) : (
        <React.Fragment>
          {psecondaryAlign === "left" && (
            <SecondaryWrapper className="secondary-wrapper" ptype={ptype}>
              {!pwithCheck ? psecondary : <PCheckbox checked={checked} defaultChecked={pchecked} />}
            </SecondaryWrapper>
          )}
          <ButtonText className="button-text" ptype={ptype} pchecked={checked.toString()}>
            {pname}
          </ButtonText>
          {psecondaryAlign === "right" && (
            <SecondaryWrapper className="secondary-wrapper" ptype={ptype}>
              {!pwithCheck ? psecondary : <PCheckbox checked={checked} defaultChecked={pchecked} />}
            </SecondaryWrapper>
          )}
        </React.Fragment>
      )}
    </CustomButton>
  );
};

PButtonSecondary.propTypes = {
  pname: PropTypes.string,
  ptype: PropTypes.oneOf(["blue", "default"]),
  psecondary: PropTypes.node,
  psecondaryAlign: PropTypes.oneOf(["left", "right"]),
  pwithCheck: PropTypes.bool,
  pchecked: PropTypes.bool,
  width: PropTypes.string,
  height: PropTypes.string,
  children: PropTypes.node,
};

PButtonSecondary.defaultProps = {
  pname: "Button",
  ptype: "default",
  pwithCheck: false,
  pchecked: false,
  height: "24px",
};

export default PButtonSecondary;
