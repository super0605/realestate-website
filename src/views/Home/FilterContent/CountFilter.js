import React, { useState } from "react";
import PropTypes from "prop-types";
import { Divider, Slider } from "antd";
import PButtonSecondary from "../../../components/PButtonSecondary";
import {
  Container,
  Content,
  Footer,
  TitleText,
  RangeValues,
  ValueElm,
  SliderWrap,
  ActionBtnGroup,
} from "./styled";

const marks = {
  0: "0",
  1: "1",
  2: "2",
  3: "3",
  4: "4",
  5: "5",
  6: "6",
  7: "7",
  8: "8",
  9: "9",
  10: {
    style: {
      color: "#f50",
    },
    label: <strong>10</strong>,
  },
};

const CountFilter = ({ hide, title, minVal, maxVal, startVal, endVal, unit, appliedFilter, valueKey }) => {
  const [rangeVal, setRangeVal] = useState(0);

  const handleChange = (val) => {
    setRangeVal(val);
  };

  const handleCancel = () => {
    setRangeVal(0);
    hide();
  };

  const handleApply = () => {
    const fData = {
      type: "static",
      valueKey: valueKey,
      value: rangeVal,
    }
    appliedFilter(fData)
    hide();
  };

  return (
    <Container>
      <Content>
        <TitleText className="filter-title">{title}</TitleText>
        <RangeValues>
          <ValueElm className="filter-value">
            {rangeVal} - {unit}
          </ValueElm>
        </RangeValues>
        <SliderWrap>
          <Slider
            marks={marks}
            step={null}
            defaultValue={0}
            min={minVal}
            max={maxVal}
            onChange={handleChange}
          />
        </SliderWrap>
      </Content>
      <Divider />
      <Footer>
        <ActionBtnGroup>
          <PButtonSecondary
            ptype="default"
            pname="CANCEL"
            psecondaryAlign="right"
            mright="16px"
            onClick={handleCancel}
          />
          <PButtonSecondary
            ptype="default"
            pname="APPLY"
            psecondaryAlign="right"
            mright="16px"
            onClick={handleApply}
          />
        </ActionBtnGroup>
      </Footer>
    </Container>
  );
};

CountFilter.propTypes = {
  hide: PropTypes.func,
  title: PropTypes.string,
  minVal: PropTypes.number,
  maxVal: PropTypes.number,
  startVal: PropTypes.number,
  endVal: PropTypes.number,
  unit: PropTypes.string,
  appliedFilter: PropTypes.func,
  valueKey: PropTypes.string,
};

export default CountFilter;
