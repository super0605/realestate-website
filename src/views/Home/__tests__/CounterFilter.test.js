import * as React from "react";
import { mount, shallow } from "enzyme";
import { Slider } from "antd";
import CountFilter from "../FilterContent/CountFilter";
import PButtonSecondary from "../../../components/PButtonSecondary";

describe("CountFilter", () => {
  let props;
  let mountedCountFilter;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedCountFilter) {
        mountedCountFilter = bShallow
        ? shallow(<CountFilter {...props} />)
        : mount(<CountFilter {...props} />);
    }
    return mountedCountFilter;
  };

  const mockHide = jest.fn();
  const appliedFilter = jest.fn();

  beforeEach(() => {
    props = {
      title: "Number of Bedrooms",
      minVal: 1,
      maxVal: 10,
      startVal: 1,
      endVal: 10,
      unit: "rooms",
      hide: mockHide,
      appliedFilter,
      valueKey: "numberOfBedrooms",
    };
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
	});

  it("should render correct title of range filter", () => {
    expect(wrapper.find(".filter-title").text()).toBe("Number of Bedrooms");
	});
	
	it("should render correct both start and end value element", () => {
    expect(wrapper.find(".filter-value").length).toBe(1);
	});

	it("should render range slider element", () => {
    expect(wrapper.find(Slider).length).toBe(1);
	});
	
	it("should render both apply and cancel button", () => {
    expect(wrapper.find(PButtonSecondary).length).toBe(2);
	});
});
