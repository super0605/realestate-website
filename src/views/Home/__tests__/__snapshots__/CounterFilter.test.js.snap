// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountFilter should render without problems 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      className="filter-title"
    >
      Number of Bedrooms
    </styled.div>
    <styled.div>
      <styled.div
        className="filter-value"
      >
        0
         - 
        rooms
      </styled.div>
    </styled.div>
    <styled.div>
      <Slider
        defaultValue={0}
        marks={
          Object {
            "0": "0",
            "1": "1",
            "10": Object {
              "label": <strong>
                10
              </strong>,
              "style": Object {
                "color": "#f50",
              },
            },
            "2": "2",
            "3": "3",
            "4": "4",
            "5": "5",
            "6": "6",
            "7": "7",
            "8": "8",
            "9": "9",
          }
        }
        max={10}
        min={1}
        onChange={[Function]}
        step={null}
        tipFormatter={[Function]}
      />
    </styled.div>
  </styled.div>
  <Divider />
  <styled.div>
    <styled.div>
      <PButtonSecondary
        height="24px"
        mright="16px"
        onClick={[Function]}
        pchecked={false}
        pname="CANCEL"
        psecondaryAlign="right"
        ptype="default"
        pwithCheck={false}
      />
      <PButtonSecondary
        height="24px"
        mright="16px"
        onClick={[Function]}
        pchecked={false}
        pname="APPLY"
        psecondaryAlign="right"
        ptype="default"
        pwithCheck={false}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
